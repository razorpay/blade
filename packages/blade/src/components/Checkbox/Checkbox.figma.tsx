import React from 'react';
import { Checkbox } from './Checkbox';
import figma from '@figma/code-connect';

/**
 * -- This file was auto-generated by `figma connect create` --
 * `props` includes a mapping from Figma properties and variants to
 * suggested values. You should update this to match the props of your
 * code component, and update the `example` function to return the
 * code example you'd like to see in Figma
 */

figma.connect(
  Checkbox,
  'https://www.figma.com/design/jubmQL9Z8V7881ayUD95ps/Blade-DSL?node-id=71951-34708&m=dev',
  {
    props: {
      helpText: figma.boolean('showHelpText'),
      children: figma.string('label'),
      size: figma.enum('size', {
        Small: 'small',
        Medium: 'medium',
        Large: 'large',
      }),
      isDisabled: figma.boolean('isDisabled'),
      isChecked: figma.boolean('isChecked'),
      isIntermediate: figma.boolean('isIntermediate'),
      validationState: figma.enum('validationState', {
        Error: 'error',
      }),
      errorText: figma.boolean('errorText'),
    },

    example: ({ children, errorText, helpText, ...props }) => (
      <Checkbox
        {...props}
        errorText={errorText} // Add error text here
        helpText={helpText} // Add help text here
      >
        {children}
      </Checkbox>
    ),
  },
);
