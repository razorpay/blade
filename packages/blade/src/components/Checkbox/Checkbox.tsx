/* eslint-disable @typescript-eslint/restrict-plus-operands */
/* eslint-disable @typescript-eslint/no-shadow */
import React from 'react';
import { useCheckboxGroupContext } from './CheckboxGroup/CheckboxGroupContext';
import { CheckboxIcon } from './CheckboxIcon';
import { useCheckbox } from './useCheckbox';
import { checkboxHoverTokens, checkboxSizes } from './checkboxTokens';
import isEmpty from '~utils/lodashButBetter/isEmpty';
import isUndefined from '~utils/lodashButBetter/isUndefined';
import { metaAttribute, MetaConstants } from '~utils/metaAttribute';
import { getStyledProps } from '~components/Box/styledProps';
import type { StyledPropsBlade } from '~components/Box/styledProps';
import BaseBox from '~components/Box/BaseBox';
import { FormHint } from '~components/Form';
import { SelectorLabel } from '~components/Form/Selector/SelectorLabel';
import { SelectorTitle } from '~components/Form/Selector/SelectorTitle';
import { SelectorSupportText } from '~components/Form/Selector/SelectorSupportText';
import { SelectorInput } from '~components/Form/Selector/SelectorInput';
import type { BladeElementRef, TestID } from '~utils/types';
import { assignWithoutSideEffects } from '~utils/assignWithoutSideEffects';
import { throwBladeError } from '~utils/logger';
import { makeSize, useTheme } from '~utils';

type OnChange = ({
  isChecked,
  event,
  value,
}: {
  isChecked: boolean;
  event?: React.ChangeEvent;
  value?: string;
}) => void;

type CheckboxProps = {
  /**
   * If `true`, The checkbox will be checked. This also makes the checkbox controlled
   * Use `onChange` to update its value
   *
   * @default false
   */
  isChecked?: boolean;
  /**
   * If `true`, the checkbox will be initially checked. This also makes the checkbox uncontrolled
   *
   * @default false
   */
  defaultChecked?: boolean;
  /**
   * The callback invoked when the checked state of the `Checkbox` changes.
   */
  onChange?: OnChange;
  /**
   * Sets the label of the checkbox
   */
  children?: React.ReactNode;
  /**
   * Help text for the checkbox
   */
  helpText?: string;
  /**
   * Error text for the checkbox
   *
   * Renders when `validationState` is set to 'error'
   */
  errorText?: string;
  /**
   * If `true`, the checkbox will be indeterminate.
   * This does not modify the isChecked property.
   *
   * @default false
   */
  isIndeterminate?: boolean;
  /**
   * The name of the input field in a checkbox
   * (Useful for form submission).
   */
  name?: string;
  /**
   * The value to be used in the checkbox input.
   * This is the value that will be returned on form submission.
   */
  value?: string;
  /**
   * If `true`, the checkbox will be disabled
   *
   * @default false
   */
  isDisabled?: boolean;
  /**
   * If `true`, the checkbox input is marked as required,
   * and `required` attribute will be added
   *
   * @default false
   */
  isRequired?: boolean;
  /**
   * If `error`, the checkbox input is marked as invalid,
   * and `invalid` attribute will be added
   */
  validationState?: 'error' | 'none';
  /**
   * Size of the checkbox
   *
   * @default "medium"
   */
  size?: 'small' | 'medium' | 'large';
  /**
   * Sets the tab-index property on checkbox element
   *
   */
  tabIndex?: number;
} & TestID &
  StyledPropsBlade;

const _Checkbox: React.ForwardRefRenderFunction<BladeElementRef, CheckboxProps> = (
  {
    defaultChecked,
    validationState,
    isChecked,
    isDisabled,
    isIndeterminate,
    isRequired,
    name,
    onChange,
    value,
    children,
    helpText,
    errorText,
    size = 'medium',
    tabIndex,
    testID,
    ...styledProps
  },
  ref,
) => {
  const groupProps = useCheckboxGroupContext();

  // ban certain props in checkbox while inside group
  const hasValidationState = !isUndefined(validationState);
  const hasName = !isUndefined(name);
  const hasDefaultChecked = !isUndefined(defaultChecked);
  const hasIsChecked = !isUndefined(isChecked);
  const hasOnChange = !isUndefined(onChange);

  if (__DEV__) {
    if (
      (hasValidationState || hasName || hasDefaultChecked || hasIsChecked || hasOnChange) &&
      !isEmpty(groupProps)
    ) {
      const props = [
        hasValidationState ? 'validationState' : undefined,
        hasName ? 'name' : undefined,
        hasDefaultChecked ? 'defaultChecked' : undefined,
        hasIsChecked ? 'isChecked' : undefined,
        hasOnChange ? 'onChange' : undefined,
      ]
        .filter(Boolean)
        .join(',');

      throwBladeError({
        message: `Cannot set \`${props}\` on <Checkbox /> when it's inside <CheckboxGroup />, Please set it on the <CheckboxGroup /> itself`,
        moduleName: 'Checkbox',
      });
    }

    // mandate value prop when using inside group
    if (!value && !isEmpty(groupProps)) {
      throw new Error(
        `[Blade Checkbox]: <CheckboxGroup /> requires that you pass unique "value" prop to each <Checkbox />
      <CheckboxGroup>
        <Checkbox value="apple">Apple</Checkbox>
        <Checkbox value="mango">Mango</Checkbox>
      </CheckboxGroup>
      `,
      );
    }
  }

  const _validationState = validationState ?? groupProps?.validationState;
  const _hasError = _validationState === 'error';
  const _isDisabled = isDisabled ?? groupProps?.isDisabled;
  const _isRequired = Boolean(
    isRequired || groupProps?.isRequired || groupProps?.necessityIndicator === 'required',
  );
  const _name = name ?? groupProps?.name;
  // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
  const _isChecked = isChecked ?? groupProps?.state?.isChecked(value!);
  const _size = groupProps.size ?? size;
  const { theme } = useTheme();
  const formHintSize = {
    small: 'medium',
    medium: 'medium',
    large: 'large',
  } as const;

  // only show error when the self validation is set to error
  // Since we don't want to show errorText inside the group
  const showSupportingText = validationState !== 'error' && helpText;

  const handleChange: OnChange = ({ isChecked, event, value }) => {
    if (isChecked) {
      groupProps?.state?.addValue(value!);
    } else {
      groupProps?.state?.removeValue(value!);
    }

    onChange?.({ isChecked, event, value });
  };

  const { state, ids, inputProps } = useCheckbox({
    defaultChecked,
    isChecked: _isChecked,
    isIndeterminate,
    hasError: _hasError,
    hasHelperText: Boolean(showSupportingText),
    isDisabled: _isDisabled,
    isRequired: _isRequired,
    name: _name,
    value,
    onChange: handleChange,
  });

  // Checkbox icon's size & margin + margin-left of SelectorTitle which is 2
  const helpTextLeftSpacing = makeSize(checkboxSizes.icon[size].width + theme.spacing[3]);

  return (
    <BaseBox
      {...metaAttribute({ name: MetaConstants.Checkbox, testID })}
      {...getStyledProps(styledProps)}
    >
      <SelectorLabel
        componentName={MetaConstants.CheckboxLabel}
        inputProps={state.isReactNative ? inputProps : {}}
      >
        <BaseBox display="flex" flexDirection="column">
          <BaseBox display="flex" flexDirection="row">
            <SelectorInput
              hoverTokens={checkboxHoverTokens}
              isChecked={state.isChecked || Boolean(isIndeterminate)}
              isDisabled={_isDisabled}
              hasError={_hasError}
              inputProps={inputProps}
              tabIndex={tabIndex}
              ref={ref}
            />
            <CheckboxIcon
              size={_size}
              isChecked={state.isChecked}
              isIndeterminate={isIndeterminate}
              isDisabled={_isDisabled}
              isNegative={_hasError}
            />
            {children ? (
              <SelectorTitle size={_size} isDisabled={_isDisabled}>
                {children}
              </SelectorTitle>
            ) : null}
          </BaseBox>
          {showSupportingText ? (
            <BaseBox marginLeft={helpTextLeftSpacing}>
              <SelectorSupportText size={_size} id={ids?.helpTextId}>
                {helpText}
              </SelectorSupportText>
            </BaseBox>
          ) : null}
        </BaseBox>
      </SelectorLabel>
      <FormHint
        size={formHintSize[_size]}
        errorText={errorText}
        errorTextId={ids?.errorTextId}
        type={validationState === 'error' ? 'error' : 'help'}
      />
    </BaseBox>
  );
};

const Checkbox = assignWithoutSideEffects(React.forwardRef(_Checkbox), {
  displayName: 'Checkbox',
});

export type { CheckboxProps };
export { Checkbox };
