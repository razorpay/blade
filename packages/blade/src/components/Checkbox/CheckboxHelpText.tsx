import React from 'react';
import BaseText from '../Typography/BaseText';

const CheckboxHelpText = ({ children }: { children: React.ReactNode }): React.ReactElement => {
  return (
    <BaseText
      color="surface.text.subtle.lowContrast"
      fontSize={50}
      lineHeight="s"
      fontStyle="italic"
      fontFamily="text"
    >
      {children}
    </BaseText>
  );
};

export { CheckboxHelpText };
