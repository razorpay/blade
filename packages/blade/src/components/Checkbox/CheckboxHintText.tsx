import React from 'react';
import getPlatformType from '../../utils/getPlatformType';
import BaseText from '../Typography/BaseText';

type CheckboxHintText = {
  id?: string;
  children: React.ReactNode;
  variant: 'help' | 'error';
};

const CheckboxHintText = ({ children, variant, id }: CheckboxHintText): React.ReactElement => {
  const isReactNative = getPlatformType() === 'react-native';

  return (
    <BaseText
      id={id}
      as={isReactNative ? undefined : 'span'}
      color={
        variant === 'error'
          ? 'feedback.text.negative.lowContrast'
          : 'surface.text.subtle.lowContrast'
      }
      fontSize={50}
      lineHeight="s"
      fontStyle="italic"
      fontFamily="text"
    >
      {children}
    </BaseText>
  );
};

export { CheckboxHintText };
