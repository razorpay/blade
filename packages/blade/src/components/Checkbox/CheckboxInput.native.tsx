/* eslint-disable react/jsx-no-useless-fragment */

// noop in react-native
const CheckboxInput = (): React.ReactElement => {
  return <></>;
};

export { CheckboxInput };
