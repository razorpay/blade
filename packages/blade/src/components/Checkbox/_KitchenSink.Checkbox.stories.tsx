import { composeStories } from '@storybook/react';
import * as checkboxStories from './Checkbox.stories';
import * as checkboxGroupStories from './CheckboxGroup.stories';
import { Box } from '~components/Box';
import { Heading } from '~components/Typography';

const allStories = [
  ...Object.values(composeStories(checkboxStories)),
  ...Object.values(composeStories(checkboxGroupStories)),
];

export const Checkbox = (): JSX.Element => {
  return (
    <Box display="flex" flexDirection="column" gap="spacing.4">
      {allStories.map((Story) => {
        return (
          <>
            <Heading>{Story.storyName}</Heading>
            <Story />
          </>
        );
      })}
    </Box>
  );
};

export default {
  title: 'Components/KitchenSink/Checkbox',
  component: Checkbox,
  parameters: {
    // enable Chromatic's snapshotting only for kitchensink
    chromatic: { disableSnapshot: false },
    options: { showPanel: false },
  },
};
