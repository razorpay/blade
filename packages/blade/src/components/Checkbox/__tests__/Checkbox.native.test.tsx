/* eslint-disable @typescript-eslint/ban-ts-comment */
/* eslint-disable @typescript-eslint/explicit-function-return-type */
import { fireEvent } from '@testing-library/react-native';
import React from 'react';
import { Text } from 'react-native';
import { Checkbox } from '../Checkbox';
import renderWithTheme from '~src/_helpers/testing/renderWithTheme.native';

describe('<Checkbox />', () => {
  it('should render checkbox with label', () => {
    const labelText = 'Remember password';
    const { toJSON, getByText } = renderWithTheme(<Checkbox>{labelText}</Checkbox>);
    expect(toJSON()).toMatchSnapshot();
    expect(getByText(labelText)).toBeDefined();
  });

  it('should render helpText', () => {
    const labelText = 'Remember password';
    const helpText = 'This has to be checked';
    const { getByText } = renderWithTheme(<Checkbox helpText={helpText}>{labelText}</Checkbox>);
    expect(getByText(helpText)).toBeDefined();
  });

  it('should set error state with validationState', () => {
    const labelText = 'Remember password';
    const errorText = 'This has to be checked';
    const { toJSON, getByRole, getByText } = renderWithTheme(
      <Checkbox validationState="error" errorText={errorText}>
        {labelText}
      </Checkbox>,
    );
    expect(toJSON()).toMatchSnapshot();

    const checkbox = getByRole('checkbox');
    expect(getByText(errorText)).toBeTruthy();
    expect(checkbox.findByProps({ accessibilityInvalid: true })).toBeTruthy();
  });

  it('should set disabled state with isDisabled', () => {
    const labelText = 'Remember password';
    const { toJSON, queryByA11yState } = renderWithTheme(
      <Checkbox isDisabled>{labelText}</Checkbox>,
    );
    expect(toJSON()).toMatchSnapshot();
    const checkbox = queryByA11yState({ disabled: true });
    expect(checkbox).toBeTruthy();
    expect(checkbox?.props?.accessibilityState).toStrictEqual({ checked: false, disabled: true });
  });

  it('should set defaultChecked', () => {
    const labelText = 'Remember password';
    const { getByRole } = renderWithTheme(<Checkbox defaultChecked>{labelText}</Checkbox>);
    const checkbox = getByRole('checkbox');
    expect(checkbox.props.accessibilityState.checked).toBeTruthy();
  });

  it('should support isChecked prop', () => {
    const labelText = 'Remember password';
    const { getByRole } = renderWithTheme(<Checkbox isChecked>{labelText}</Checkbox>);
    const checkbox = getByRole('checkbox');

    expect(checkbox.props.accessibilityState.checked).toBeTruthy();
    // should not toggle
    fireEvent.press(checkbox);
    expect(checkbox.props.accessibilityState.checked).toBeTruthy();
  });

  test('user should be able to toggle checkbox', () => {
    const labelText = 'Remember password';
    const { getByRole } = renderWithTheme(<Checkbox>{labelText}</Checkbox>);
    const checkbox = getByRole('checkbox');

    expect(checkbox.props.accessibilityState.checked).toBeFalsy();
    fireEvent.press(checkbox);
    expect(checkbox.props.accessibilityState.checked).toBeTruthy();
    fireEvent.press(checkbox);
    expect(checkbox.props.accessibilityState.checked).toBeFalsy();
  });

  it('should support uncontrolled state', () => {
    const checkFn = jest.fn();
    const eventData = {
      nativeEvent: {
        pageX: 20,
        pageY: 30,
      },
    };
    const labelText = 'Remember password';
    const { getByRole } = renderWithTheme(
      <Checkbox defaultChecked={true} onChange={checkFn}>
        {labelText}
      </Checkbox>,
    );
    const checkbox = getByRole('checkbox');

    expect(checkbox.props.accessibilityState.checked).toBeTruthy();
    expect(checkFn).not.toBeCalled();
    fireEvent.press(checkbox, eventData);
    expect(checkbox.props.accessibilityState.checked).toBeFalsy();
    expect(checkFn).toBeCalledWith(
      expect.objectContaining({
        isChecked: false,
        value: undefined,
        event: eventData,
      }),
    );
    fireEvent.press(checkbox, eventData);
    expect(checkbox.props.accessibilityState.checked).toBeTruthy();
    expect(checkFn).toBeCalledWith(
      expect.objectContaining({
        isChecked: true,
        value: undefined,
        event: eventData,
      }),
    );
  });

  it('should support controlled state', () => {
    const labelText = 'Remember password';
    const Example = () => {
      const [checked, setChecked] = React.useState(false);
      return (
        <>
          <Checkbox isChecked={checked} onChange={({ isChecked }) => setChecked(isChecked)}>
            {labelText}
          </Checkbox>
          <Text testID="state">{checked ? 'checked' : 'unchecked'}</Text>
        </>
      );
    };
    const { getByTestId, getByRole } = renderWithTheme(<Example />);
    const checkbox = getByRole('checkbox');

    expect(getByTestId('state').children[0]).toBe('unchecked');
    fireEvent.press(checkbox);
    expect(getByTestId('state').children[0]).toBe('checked');
    fireEvent.press(checkbox);
    expect(getByTestId('state').children[0]).toBe('unchecked');
  });

  it('should expose native element methods via ref', () => {
    let refValue = null;
    const Example = (): React.ReactElement => {
      const ref = React.useRef<HTMLInputElement>(null);
      return (
        <Checkbox
          ref={(value) => {
            console.log(value);
            // @ts-expect-error
            ref.current = value;
            refValue = value;
          }}
        >
          Agree
        </Checkbox>
      );
    };

    renderWithTheme(<Example />);
    expect(refValue).toBeNull();
  });
});
