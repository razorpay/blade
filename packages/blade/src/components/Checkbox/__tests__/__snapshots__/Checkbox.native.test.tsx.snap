// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> should render checkbox with label 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    data-blade-component="checkbox"
    style={
      [
        {},
      ]
    }
  >
    <View
      accessibilityInvalid={false}
      accessibilityRequired={false}
      accessibilityRole="checkbox"
      accessibilityState={
        {
          "busy": undefined,
          "checked": false,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "display": "flex",
            "flexDirection": "row",
            "marginBottom": 2,
            "marginTop": 2,
          },
        ]
      }
    >
      <View
        data-blade-component="base-box"
        display="flex"
        flexDirection="column"
        style={
          [
            {
              "display": "flex",
              "flexDirection": "column",
            },
          ]
        }
      >
        <View
          data-blade-component="base-box"
          display="flex"
          flexDirection="row"
          style={
            [
              {
                "display": "flex",
                "flexDirection": "row",
              },
            ]
          }
        >
          <View
            data-blade-component="checkbox-icon-wrapper"
            isChecked={false}
            isNegative={false}
            size="medium"
            style={
              [
                {
                  "alignItems": "center",
                  "backgroundColor": "hsla(0, 0%, 100%, 0)",
                  "borderColor": "hsla(211, 27%, 76%, 1)",
                  "borderRadius": 2,
                  "borderStyle": "solid",
                  "borderWidth": 1.5,
                  "display": "flex",
                  "flexShrink": 0,
                  "height": 16,
                  "justifyContent": "center",
                  "marginBottom": 2,
                  "marginLeft": 2,
                  "marginRight": 2,
                  "marginTop": 2,
                  "position": "relative",
                  "width": 16,
                },
              ]
            }
          />
          <View
            data-blade-component="base-box"
            style={
              [
                {
                  "marginLeft": 4,
                },
              ]
            }
          />
          <Text
            accessible={true}
            color="surface.text.gray.subtle"
            data-blade-component="text"
            fontFamily="text"
            fontSize={100}
            fontStyle="normal"
            fontWeight="regular"
            lineHeight={100}
            style={
              [
                {
                  "color": "hsla(211, 26%, 34%, 1)",
                  "fontFamily": "Inter",
                  "fontSize": 14,
                  "fontStyle": "normal",
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 20,
                  "marginBottom": 0,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 0,
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 0,
                  "paddingTop": 0,
                  "textDecorationLine": "none",
                },
              ]
            }
          >
            Remember password
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`<Checkbox /> should set disabled state with isDisabled 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    data-blade-component="checkbox"
    style={
      [
        {},
      ]
    }
  >
    <View
      accessibilityInvalid={false}
      accessibilityRequired={false}
      accessibilityRole="checkbox"
      accessibilityState={
        {
          "busy": undefined,
          "checked": false,
          "disabled": true,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "display": "flex",
            "flexDirection": "row",
            "marginBottom": 2,
            "marginTop": 2,
          },
        ]
      }
    >
      <View
        data-blade-component="base-box"
        display="flex"
        flexDirection="column"
        style={
          [
            {
              "display": "flex",
              "flexDirection": "column",
            },
          ]
        }
      >
        <View
          data-blade-component="base-box"
          display="flex"
          flexDirection="row"
          style={
            [
              {
                "display": "flex",
                "flexDirection": "row",
              },
            ]
          }
        >
          <View
            data-blade-component="checkbox-icon-wrapper"
            isChecked={false}
            isDisabled={true}
            isNegative={false}
            size="medium"
            style={
              [
                {
                  "alignItems": "center",
                  "backgroundColor": "hsla(0, 0%, 100%, 0)",
                  "borderColor": "hsla(214, 40%, 92%, 1)",
                  "borderRadius": 2,
                  "borderStyle": "solid",
                  "borderWidth": 1.5,
                  "display": "flex",
                  "flexShrink": 0,
                  "height": 16,
                  "justifyContent": "center",
                  "marginBottom": 2,
                  "marginLeft": 2,
                  "marginRight": 2,
                  "marginTop": 2,
                  "position": "relative",
                  "width": 16,
                },
              ]
            }
          />
          <View
            data-blade-component="base-box"
            style={
              [
                {
                  "marginLeft": 4,
                },
              ]
            }
          />
          <Text
            accessible={true}
            color="surface.text.gray.disabled"
            data-blade-component="text"
            fontFamily="text"
            fontSize={100}
            fontStyle="normal"
            fontWeight="regular"
            lineHeight={100}
            style={
              [
                {
                  "color": "hsla(211, 20%, 52%, 0.32)",
                  "fontFamily": "Inter",
                  "fontSize": 14,
                  "fontStyle": "normal",
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 20,
                  "marginBottom": 0,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 0,
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 0,
                  "paddingTop": 0,
                  "textDecorationLine": "none",
                },
              ]
            }
          >
            Remember password
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`<Checkbox /> should set error state with validationState 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    data-blade-component="checkbox"
    style={
      [
        {},
      ]
    }
  >
    <View
      accessibilityErrorMessage="checkbox-25-errortext-27"
      accessibilityInvalid={true}
      accessibilityRequired={false}
      accessibilityRole="checkbox"
      accessibilityState={
        {
          "busy": undefined,
          "checked": false,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "display": "flex",
            "flexDirection": "row",
            "marginBottom": 2,
            "marginTop": 2,
          },
        ]
      }
    >
      <View
        data-blade-component="base-box"
        display="flex"
        flexDirection="column"
        style={
          [
            {
              "display": "flex",
              "flexDirection": "column",
            },
          ]
        }
      >
        <View
          data-blade-component="base-box"
          display="flex"
          flexDirection="row"
          style={
            [
              {
                "display": "flex",
                "flexDirection": "row",
              },
            ]
          }
        >
          <View
            data-blade-component="checkbox-icon-wrapper"
            isChecked={false}
            isNegative={true}
            size="medium"
            style={
              [
                {
                  "alignItems": "center",
                  "backgroundColor": "hsla(0, 0%, 100%, 0)",
                  "borderColor": "hsla(4, 74%, 49%, 1)",
                  "borderRadius": 2,
                  "borderStyle": "solid",
                  "borderWidth": 1.5,
                  "display": "flex",
                  "flexShrink": 0,
                  "height": 16,
                  "justifyContent": "center",
                  "marginBottom": 2,
                  "marginLeft": 2,
                  "marginRight": 2,
                  "marginTop": 2,
                  "position": "relative",
                  "width": 16,
                },
              ]
            }
          />
          <View
            data-blade-component="base-box"
            style={
              [
                {
                  "marginLeft": 4,
                },
              ]
            }
          />
          <Text
            accessible={true}
            color="surface.text.gray.subtle"
            data-blade-component="text"
            fontFamily="text"
            fontSize={100}
            fontStyle="normal"
            fontWeight="regular"
            lineHeight={100}
            style={
              [
                {
                  "color": "hsla(211, 26%, 34%, 1)",
                  "fontFamily": "Inter",
                  "fontSize": 14,
                  "fontStyle": "normal",
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 20,
                  "marginBottom": 0,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 0,
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 0,
                  "paddingTop": 0,
                  "textDecorationLine": "none",
                },
              ]
            }
          >
            Remember password
          </Text>
        </View>
      </View>
    </View>
    <View
      data-blade-component="base-box"
      style={
        [
          {
            "marginTop": 4,
          },
        ]
      }
    >
      <View
        alignItems="center"
        data-blade-component="base-box"
        display="flex"
        flexDirection="row"
        style={
          [
            {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
            },
          ]
        }
      >
        <RNSVGSvgView
          accessibilityElementsHidden={true}
          align="xMidYMid"
          bbHeight="12px"
          bbWidth="12px"
          data-blade-component="icon"
          fill="none"
          focusable={false}
          height="12px"
          importantForAccessibility="no-hide-descendants"
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              [
                {},
              ],
              {
                "flex": 0,
                "height": 12,
                "width": 12,
              },
            ]
          }
          vbHeight={24}
          vbWidth={24}
          width="12px"
        >
          <RNSVGGroup
            fill={null}
            propList={
              [
                "fill",
              ]
            }
          >
            <RNSVGPath
              d="M12 11C12.5523 11 13 11.4477 13 12V16C13 16.5523 12.5523 17 12 17C11.4477 17 11 16.5523 11 16V12C11 11.4477 11.4477 11 12 11Z"
              fill={
                {
                  "payload": 4292422944,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M12 7C11.4477 7 11 7.44772 11 8C11 8.55228 11.4477 9 12 9H12.01C12.5623 9 13.01 8.55228 13.01 8C13.01 7.44772 12.5623 7 12.01 7H12Z"
              fill={
                {
                  "payload": 4292422944,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              clipRule={0}
              d="M1 12C1 5.92487 5.92487 1 12 1C18.0751 1 23 5.92487 23 12C23 18.0751 18.0751 23 12 23C5.92487 23 1 18.0751 1 12ZM12 3C7.02944 3 3 7.02944 3 12C3 16.9706 7.02944 21 12 21C16.9706 21 21 16.9706 21 12C21 7.02944 16.9706 3 12 3Z"
              fill={
                {
                  "payload": 4292422944,
                  "type": 0,
                }
              }
              fillRule={0}
              propList={
                [
                  "fill",
                  "fillRule",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
        <View
          data-blade-component="base-box"
          style={
            [
              {
                "marginRight": 4,
              },
            ]
          }
        />
        <Text
          accessible={true}
          color="feedback.text.negative.intense"
          data-blade-component="text"
          fontFamily="text"
          fontSize={50}
          fontStyle="italic"
          fontWeight="regular"
          lineHeight={50}
          style={
            [
              {
                "color": "hsla(4, 74%, 49%, 1)",
                "fontFamily": "Inter",
                "fontSize": 11,
                "fontStyle": "italic",
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 16,
                "marginBottom": 0,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 0,
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 0,
                "textDecorationLine": "none",
              },
            ]
          }
        >
          This has to be checked
        </Text>
      </View>
    </View>
  </View>
</View>
`;
